corehq/apps/geospatial/views.py

```diff
@@ -293,12 +295,12 @@ def _get_paginated_users_without_gps(domain, page, limit, query):
     user_docs = get_docs(CommCareUser.get_db(), keys=user_ids_page)
     user_data = []
     for user_doc in user_docs:
-        lat, lon = ('', '')
-        if location_prop_name in user_doc['user_data']:
+        try:
             gps_location_prop = user_doc['user_data'][location_prop_name]
-            gps_data = gps_location_prop.split(' ')
-            if len(gps_data) > 1:
-                lat, lon = (gps_data[0], gps_data[1])
+            geo_point = GeoPoint.from_string(gps_location_prop, flexible=True)
+            lat, lon = (str(geo_point.latitude), str(geo_point.longitude))
+        except (KeyError, BadValueError):
+            lat, lon = ('', '')
```

Comment:
Nice! can we avoid the code duplication here?

Reply:
[95b33e9](https://github.com/dimagi/commcare-hq/pull/33420/commits/95b33e94633b36813a67c02f61229e24f8081c67)